enhancements/update/dont-require-registry-during-boot-and-upgrade.md

```diff
@@ -0,0 +1,201 @@
+---
+title: dont-require-registry-during-boot-and-upgrade
+authors:
+- "@jhernand"
+reviewers:
+- "@avishayt"
+- "@danielerez"
+- "@mrunalp"
+- "@nmagnezi"
+- "@oourfali"
+approvers:
+- "@sdodson"
+- "@zaneb"
+- "@LalatenduMohanty"
+api-approvers:
+- "@sdodson"
+- "@zaneb"
+- "@deads2k"
+- "@JoelSpeed"
+creation-date: 2023-09-21
+last-updated: 2023-09-21
+tracking-link:
+- https://issues.redhat.com/browse/RFE-4482
+see-also:
+- https://issues.redhat.com/browse/OCPBUGS-13219
+- https://github.com/openshift/enhancements/pull/1481
+- https://github.com/openshift/cluster-network-operator/pull/1803
+replaces: []
+superseded-by: []
+---
+
+# Don't require registry during boot and upgrade
+
+## Summary
+
+Ensure that clusters don't require a registry server to boot or upgrade when
+all the required images have already been pulled.
+
+## Motivation
+
+Currently during reboots and upgrades clusters may need to contact the image
+registry servers, even if the images have already been pulled. This complicates
+things for clusters that are completely disconnected or that have an slow or
+unreliable connection to the image registry servers.
+
+### User Stories
+
+#### Boot without registry
+
+As the administrator of a cluster that has all the required images already
+pulled in all the nodes, I want to be able to reboot it without requiring
+access to a registry server.
+
+### Upgrade without registry
+
+As the administrator of a cluster that has all the required images already
+pulled in all the nodes, I want to be able to upgrade it without requiring
+access to a registry server.
+
+### Goals
+
+Ensure that clusters don't require a registry server to boot or upgrade when
```

Comment:
it may not require one during boot/upgrade, but the content had to get to the node from somewhere (during the pull+pin phase), is that expected to be somewhere other than a registry?

Reply:
No, in the context of this EP (and also #1481) we are assuming that images come from a registry.

In the now closed #1432 we proposed the use of an "upgrade bundle", but that is completely out of the scope here.

Reply:
> Reducing the chances of failed upgrades is one of the goals of https://github.com/openshift/enhancements/pull/1481

ack, might be worth making that clear somewhere in the EP such as a non-goal of "eliminating the requirement that a registry exists(we're just eliminating the requirement that it be available/used during bootstrap+upgrade steps)"

Reply:
OK, will do.

Reply:
Done.

Reply:
Reading through I understand how we could test hosts rebooting, and I guess how we could test new nodes booting via tests. But I don't get how the new node scaleup would actually work successfully, are we saying that we expect the boot image to have all container images burned in?

Reply:
Or if we're specifically not addressing a scaled up node switch all instances of "boot" to "reboot" so that it's clear we expect the node to be able to retrieve images at least once before the registry can go inaccessible.

Reply:
When I say "boot" I mean "reboot", I will fix that.

Reply:
@sdodson the document now says "reboot" everywhere, please take another look.